Lib/test/libregrtest/main.py

```diff
@@ -309,6 +309,8 @@ def run_tests_sequentially(self, runtests) -> None:
         else:
             tracer = None
 
+        save_modules = set(sys.modules)
```

Comment:
I would prefer to move this logic inside `libregrtest/single.py`, in the `run_single_test()` function.

Reply:
I did it, but it is only needed in single-process run. In multi-process run it only adds an overhead. So I moved it back.